Frameworks/UIKit/UIScrollView.mm

```diff
@@ -169,87 +162,29 @@ - (id)_dManipGestureCallback:(_UIDMPanGestureRecognizer*)gesture {
-
-    // Grab the content canvas for our subviews - we created it in createXamlElement
-    _contentCanvas = XamlControls::GetFrameworkElementSublayerCanvasProperty(_scrollViewer);
-
-    // Grab the image for our rendered content - we created it in createXamlElement
-    _contentImage = XamlControls::GetFrameworkElementLayerContentProperty(_scrollViewer);
-
-    // creating and build 3 X 3 content grid
-    _contentGrid = [WXCGrid make];
-    WXCColumnDefinition* column0 = [WXCColumnDefinition make];
-    WXCColumnDefinition* column1 = [WXCColumnDefinition make];
-    WXCColumnDefinition* column2 = [WXCColumnDefinition make];
-
-    column0.width = [WXGridLengthHelper fromValueAndType:0 type:WXGridUnitTypeAuto];
-    column1.width = [WXGridLengthHelper fromValueAndType:0 type:WXGridUnitTypeAuto];
-    column2.width = [WXGridLengthHelper fromValueAndType:0 type:WXGridUnitTypeAuto];
-    WXCRowDefinition* row0 = [WXCRowDefinition make];
-    WXCRowDefinition* row1 = [WXCRowDefinition make];
-    WXCRowDefinition* row2 = [WXCRowDefinition make];
-    row0.height = [WXGridLengthHelper fromValueAndType:0 type:WXGridUnitTypeAuto];
-    row1.height = [WXGridLengthHelper fromValueAndType:0 type:WXGridUnitTypeAuto];
-    row2.height = [WXGridLengthHelper fromValueAndType:0 type:WXGridUnitTypeAuto];
-
-    [_contentGrid.columnDefinitions addObject:column0];
-    [_contentGrid.columnDefinitions addObject:column1];
-    [_contentGrid.columnDefinitions addObject:column2];
-    [_contentGrid.rowDefinitions addObject:row0];
-    [_contentGrid.rowDefinitions addObject:row1];
-    [_contentGrid.rowDefinitions addObject:row2];
-
-    // put insets into top/right/bottom/left in Grid
-    [_contentGrid.children addObject:_topInset];
-    [WXCGrid setColumn:_topInset value:1];
-    [WXCGrid setRow:_topInset value:0];
-
-    [_contentGrid.children addObject:_rightInset];
-    [WXCGrid setColumn:_rightInset value:2];
-    [WXCGrid setRow:_rightInset value:1];
-
-    [_contentGrid.children addObject:_bottomInset];
-    [WXCGrid setColumn:_bottomInset value:1];
-    [WXCGrid setRow:_bottomInset value:2];
-
-    [_contentGrid.children addObject:_leftInset];
-    [WXCGrid setColumn:_leftInset value:0];
-    [WXCGrid setRow:_leftInset value:1];
-
-    // put content canvas in the middle
-    [_contentGrid.children addObject:_contentCanvas];
-    [WXCGrid setColumn:_contentCanvas value:1];
-    [WXCGrid setRow:_contentCanvas value:1];
-
-    // set contentGrid as the content of the scrollviewer
-    _scrollViewer.content = _contentGrid;
-
-    // by default enable chaining
-    _scrollViewer.isVerticalScrollChainingEnabled = YES;
-    _scrollViewer.isHorizontalScrollChainingEnabled = YES;
-
-    // by default disable railing
-    _scrollViewer.isHorizontalRailEnabled = NO;
-    _scrollViewer.isVerticalRailEnabled = NO;
+    // Grab the content canvas for our subviews
+    _contentCanvas = XamlControls::GetFrameworkElementSublayerCanvasProperty(scrollViewGrid);
```

Comment:
I see. i believe that is why the attached property line is required! 

---
In reply to: [106708779](https://github.com/Microsoft/WinObjC/pull/2257#discussion_r106708779) [](ancestors = 106708779)